internal/core/src/exec/operator/ElementFilterBitsNode.cpp

```diff
@@ -0,0 +1,217 @@
+// limitations under the License.
+
+#include "ElementFilterBitsNode.h"
+#include "common/Tracer.h"
+#include "fmt/format.h"
+
+#include "monitor/Monitor.h"
+
+namespace milvus {
+namespace exec {
+
+PhyElementFilterBitsNode::PhyElementFilterBitsNode(
+    int32_t operator_id,
+    DriverContext* driverctx,
+    const std::shared_ptr<const plan::ElementFilterBitsNode>&
+        element_filter_bits_node)
+    : Operator(driverctx,
+               DataType::NONE,
+               operator_id,
+               "element_filter_bits_plan_node",
+               "PhyElementFilterBitsNode"),
+      struct_name_(element_filter_bits_node->struct_name()) {
+    ExecContext* exec_context = operator_context_->get_exec_context();
+    query_context_ = exec_context->get_query_context();
+
+    // Build expression set from element-level expression
+    std::vector<expr::TypedExprPtr> exprs;
+    exprs.emplace_back(element_filter_bits_node->element_filter());
+    element_exprs_ = std::make_unique<ExprSet>(exprs, exec_context);
+}
+
+void
+PhyElementFilterBitsNode::AddInput(RowVectorPtr& input) {
+    input_ = std::move(input);
+}
+
+RowVectorPtr
+PhyElementFilterBitsNode::GetOutput() {
+    if (is_finished_ || input_ == nullptr) {
+        return nullptr;
+    }
+
+    DeferLambda([&]() { is_finished_ = true; });
+
+    tracer::AutoSpan span(
+        "PhyElementFilterBitsNode::GetOutput", tracer::GetRootSpan(), true);
+
+    std::chrono::high_resolution_clock::time_point start_time =
+        std::chrono::high_resolution_clock::now();
+    std::chrono::high_resolution_clock::time_point step_time;
+
+    // Step 1: Get array offsets
+    auto segment = query_context_->get_segment();
+    auto field_meta = milvus::FindFirstArrayFieldInStruct(segment->get_schema(),
+                                                          struct_name_);
+    auto field_id = field_meta.get_id();
+    auto array_offsets = segment->GetArrayOffsets(field_id);
+    if (array_offsets == nullptr) {
+        ThrowInfo(ErrorCode::UnexpectedError,
+                  "IArrayOffsets not found for field {}",
+                  field_id.get());
+    }
+    query_context_->set_array_offsets(array_offsets);
+    auto [first_elem, _] =
+        array_offsets->ElementIDRangeOfRow(query_context_->get_active_count());
+    query_context_->set_active_element_count(first_elem);
+
+    // Step 2: Prepare doc bitset
+    auto col_input = GetColumnVector(input_);
+    TargetBitmapView doc_bitset(col_input->GetRawData(), col_input->size());
+    TargetBitmapView doc_bitset_valid(col_input->GetValidRawData(),
+                                      col_input->size());
+    doc_bitset.flip();
+
+    // Step 3: Convert doc bitset to element offsets
+    FixedVector<int32_t> element_offsets =
+        DocBitsetToElementOffsets(doc_bitset);
+
+    // Step 4: Evaluate element expression
+    auto [expr_result, valid_expr_result] =
+        EvaluateElementExpression(element_offsets);
+
```

Comment:
would it be possible to combine `DocBitsetToElementOffsets` and `EvaluateElementExpression` so we can eliminate the immediate `FixedVector<int32_t> element_offsets`?

Reply:
I think this is clearer? The vector element_offsets actually does not sustain much shorter as step 5 finishes quickly.